tests/components/add-elements-panel/add-elements-panel.spec.tsx

```diff
@@ -54,6 +54,51 @@ describe("AddElementsPanel", () => {
             });
         });
     });
+
+    describe("search elements by name", () => {
+        it("should show search results and cross button on search bar when input is given", async () => {
+            renderComponent();
+            const magnifyingGlass = screen.getByTestId("maginfying-glass");
+            fireEvent.click(magnifyingGlass);
+            const searchInput = screen.getByTestId("input");
+            fireEvent.change(searchInput, { target: { value: "email" } });
+
+            await waitFor(() => {
+                const getEmailAddressButton =
+                    screen.getByText(/email address/i);
+                const getCrossButton = screen.getAllByRole("button");
+                expect(getEmailAddressButton).toBeInTheDocument();
+                expect(getCrossButton[0]).toBeInTheDocument();
+            });
+        });
+        it("should be able to clear input when clicking on the cross button", async () => {
+            renderComponent();
+            const magnifyingGlass = screen.getByTestId("maginfying-glass");
+            fireEvent.click(magnifyingGlass);
+            const searchInput = screen.getByTestId("input");
+            fireEvent.change(searchInput, { target: { value: "email" } });
+            const getCrossButton = screen.getAllByRole("button");
+            fireEvent.click(getCrossButton[0]);
+
+            await waitFor(() => {
+                expect(searchInput).toHaveValue("");
+            });
+        });
+
+        it("should show results not found when input given is not an element name", async () => {
+            renderComponent();
+            const magnifyingGlass = screen.getByTestId("maginfying-glass");
+            fireEvent.click(magnifyingGlass);
+            const searchInput = screen.getByTestId("input");
+            fireEvent.change(searchInput, { target: { value: "abc" } });
+
+            await waitFor(() => {
+                const getNoResultsFoundContent =
+                    screen.getByText("No results found.");
+                expect(getNoResultsFoundContent).toBeInTheDocument();
+            });
+        });
```

Comment:
add one more test to assert a category is to be hidden if it has no matching children

Reply:
I am actually referring to this scenario where `Option group` category is hidden since it has no matching children. but I realise you may be stuck because we dont support the `Option group` yet, so may I suggest to:
- still fix this test to assert based on that scenario
- skip this test and leave a comment that it cant be run because `Option group` category is not ready yet and to unskip it when that category is added

<img width="368" alt="image" src="https://github.com/LifeSG/web-form-builder/assets/60728851/bb3f92c4-fec6-4943-bf4b-bb7d694178c9">

Reply:
has been updated in the latest changes

Reply:
by skipping I meant using `it.skip()` instead of commenting the test case out